server/controllers/admin.py

```diff
@@ -124,6 +126,17 @@ def assignment(cid, aid):
     if assgn.course != current_course:
         return abort(401)
 
+    assgn.name = form.name._value()
```

Comment:
You only have to use `form.validate_on_submit()`. It'll check the request method is POST for you. 
Docs: https://flask-wtf.readthedocs.org/en/latest/quickstart.html#validating-forms